spec/currency_spec.rb

```diff
@@ -37,6 +37,66 @@
     it "raises when the currency is nil" do
       expect { Money::Currency.new(nil) }.to raise_error(Money::Currency::UnknownCurrency)
     end
+    
+    it "looks up crypto currencies when enabled" do
+      old_config = Money.config
+      Money.config = Money::Config.new.tap do |config|
+        config.crypto_currencies = true
+      end
+      
+      allow(Money::Currency).to receive(:currencies).and_return({})
+      crypto_currencies = { 
+        "usdc" => { 
+          "iso_code" => "USDC", 
+          "name" => "USD Coin",
+          "symbol" => "USDC",
+          "disambiguate_symbol" => "USDC",
+          "subunit_to_unit" => 100,
+          "smallest_denomination" => 1,
+          "decimal_mark" => "."
+        } 
+      }
+      allow(Money::Currency).to receive(:crypto_currencies).and_return(crypto_currencies)
+      
+      currency = Money::Currency.new('USDC')
+      expect(currency.iso_code).to eq('USDC')
+      expect(currency.symbol).to eq('USDC')
+      
+      Money.config = old_config
```

Comment:
question: Does this been to be in an `ensure` block?

If the test fails I'm assuming the config won't be restored to the previous state and the custom config might affect other tests. I'm guessing we need something like:

```ruby
old_config = Money.config

begin
  # the test
ensure
  Money.config = old_config
end
```

I wonder if it's worth adding a `with_config` method [here](https://github.com/Shopify/money/blob/4377608dd60f42d58fc84e4f0d329ad65da6d697/lib/money/money.rb#L52)?

```ruby
def with_config(config)
  old_config = Money.config
  Money.config = config
  yield
ensure
  Money.config = old_config
end
```